wgpu/src/lib.rs

```diff
@@ -1575,10 +1672,49 @@ pub struct ComputePipelineDescriptor<'a> {
     /// The name of the entry point in the compiled shader. There must be a function with this name
     /// and no return value in the shader.
     pub entry_point: &'a str,
+    /// The pipeline cache to use when creating this pipeline.
+    pub cache: Option<&'a PipelineCache>,
 }
 #[cfg(send_sync)]
 static_assertions::assert_impl_all!(ComputePipelineDescriptor<'_>: Send, Sync);
 
+/// Describes a pipeline cache which reuses data from a previous run.
+///
+/// For use with [`Device::create_pipeline_cache_init`].
+///
+/// This type is unique to the Rust API of `wgpu`.
+#[derive(Clone, Debug)]
+pub struct PipelineCacheInitDescriptor<'a> {
```

Comment:
I feel like these two endpoints can be combined into one, with data being an option of a struct. This would be a lot cleaner than needing two entirely separate entry points.

Reply:
We still want it to be unsafe, right?

This would be equivalent to the bounds check policy unchecked constructor?

Reply:
Yeah, I think it is perfectly reasonable to have all pipeline caches be unsafe to construct - constructing an empty one just isn't enough of a use case for it to really bother me.